Add more text modifiers for CanvasRenderingContext2D (#23)
GoogleChromeLabs/puppy-content: content/en-US/api/textRendering.md

---
recipe: api-interface
title: 'CanvasRenderingContext2D.textRendering'
mdn_url: /en-US/docs/Web/API/CanvasRenderingContext2D/textRendering
specifications: [[specURL]]#[[HeadingAnchor]]
browser_compatibility: api.CanvasRenderingContext2D.textRendering
---

**When this feature ships, the content below will live on MDN under
[developer.mozilla.org/en-US/docs/Web/CanvasRenderingContext2D/textRendering](https://developer.mozilla.org/en-US/docs/Web/CanvasRenderingContext2D/textRendering).**

## Description

The `textRendering` property of `CanvasRenderingContext2D` provides information
to the rendering engine about what to optimize for when rendering text.

The default value is `auto`, where the browser makes trade-offs among speed, 

> legibility, and geometric precision.

Reviewer: ```suggestion
```

Reviewer: The default value is typically listed with the possible values.

Author: corrected.

## Syntax

> `CanvasRenderingContext2D.textRendering = "auto" || "optimizeSpeed" || "optimizeLegibility" || "geometricPrecision"`

Reviewer: ```suggestion
`CanvasRenderingContext2D.textRendering = tetRendering`
```

`var textRendering = CanvasRenderingContext2D.textRendering`

### Value

> A `string` representing current `textRendering` value. The possible values are:

Reviewer: ```suggestion
A `string` representing the current `textRendering` value. The possible values are:
```

Author: corrected.

> #### Options

Reviewer: ```suggestion
```

`"auto"`
The browser makes educated guesses about when to optimize for speed, legibility, 

> and geometric precision while drawing text. 

Reviewer: ```suggestion
and geometric precision while drawing text.  This is the default.
```

Author: corrected.

`"optimizeSpeed"`
The browser emphasizes rendering speed over legibility and geometric precision 
when drawing text. It disables kerning and ligatures.

`"optimizeLegibility"`
The browser emphasizes legibility over rendering speed and geometric precision.
This enables kerning and optional ligatures.

`"geometricPrecision"`
The browser emphasizes geometric precision over rendering speed and legibility. 

> Certain aspects of fonts, don't scale linearly(e.g. kerning). Geometric Precision 

Reviewer: ```suggestion
Certain aspects of fonts, don't scale linearly (e.g. kerning). Geometric Precision 
```

Author: corrected

can make text using those fonts look good.

Another usage of `geometricPrecision`is that it allows user to scale text

> fluidly, it understands float value for font size.

Reviewer: ```suggestion
an make text using those fonts look good. This value also allows developers to
scale text fluidly and it accepts float values for font size.
```

Author: corrected.

---
recipe: api-interface
title: 'CanvasRenderingContext2D.textRendering'
mdn_url: /en-US/docs/Web/API/CanvasRenderingContext2D/textRendering
specifications: [[specURL]]#[[HeadingAnchor]]
browser_compatibility: api.CanvasRenderingContext2D.textRendering
---

> 

Reviewer 2: Whitespace? Maybe I'm wrong

Author: removed!

## Description

The `textRendering` property of `CanvasRenderingContext2D` provides information
to the rendering engine about what to optimize for when rendering text.

The default value is `auto`, where the browser makes trade-offs among speed, 
legibility, and geometric precision.

## Syntax

`CanvasRenderingContext2D.textRendering = "auto" || "optimizeSpeed" || "optimizeLegibility" || "geometricPrecision"`
`var textRendering = CanvasRenderingContext2D.textRendering`

### Value
A `string` representing current `textRendering` value. The possible values are:

#### Options
`"auto"`
The browser makes educated guesses about when to optimize for speed, legibility, 
and geometric precision while drawing text. 

`"optimizeSpeed"`
The browser emphasizes rendering speed over legibility and geometric precision 
when drawing text. It disables kerning and ligatures.

`"optimizeLegibility"`
The browser emphasizes legibility over rendering speed and geometric precision.
This enables kerning and optional ligatures.

`"geometricPrecision"`
The browser emphasizes geometric precision over rendering speed and legibility. 

> Certain aspects of fonts, don't scale linearly(ex: Kerning). Geometric Precision 

Reviewer 2: linearly(ex: Kerning) -> linearly (e.g. kerning)
Geometric Precision -> Geometric precision

Author: done

can make text using those fonts look good.

Another usage of `geometricPrecision`is that it allows user to scale text
fluidly, it understands float value for font size.

### Example

> This example demonstrates the various `textRendering` property values.

Reviewer 2: colon "values:"

Author: done.
